sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -285,35 +285,37 @@ inline pi_result ur2piDeviceInfoValue(ur_device_info_t ParamName,
     return Value.convertBitSet<ur_device_affinity_domain_flag_t,
                                pi_device_affinity_domain>(ConvertFunc);
   } else if (ParamName == UR_DEVICE_INFO_PARTITION_TYPE) {
-    auto ConvertFunc = [](ur_device_partition_property_t UrValue) {
+    auto ConvertFunc = [](ur_device_partition_t UrValue) {
       switch (UrValue) {
       case UR_DEVICE_PARTITION_BY_AFFINITY_DOMAIN:
         return PI_DEVICE_PARTITION_BY_AFFINITY_DOMAIN;
       case UR_DEVICE_PARTITION_BY_CSLICE:
         return PI_EXT_INTEL_DEVICE_PARTITION_BY_CSLICE;
-      case (ur_device_partition_property_t)
+      case (ur_device_partition_t)
           UR_DEVICE_AFFINITY_DOMAIN_FLAG_NEXT_PARTITIONABLE:
         return (pi_device_partition_property)
             PI_DEVICE_AFFINITY_DOMAIN_NEXT_PARTITIONABLE;

```

Comment:
There is a warning
```
sycl/plugins/unified_runtime/pi2ur.hpp:294:7: warning: case value ‘32’ not in enumerated type ‘ur_device_partition_t’ [-Wswitch]                                                                   
  294 |       case (ur_device_partition_t)                                                                                                                                                                                      
      |       ^~~~                                                                                                                                                                                                              
[3/46] Building CXX object tools/sycl/plugins/level_zero/CMakeFiles/pi_level_zero.dir/pi_level_zero.cpp.o                                                                                                                       
In file included from sycl/plugins/level_zero/pi_level_zero.hpp:30,                                                                                                                                
                 from sycl/plugins/level_zero/pi_level_zero.cpp:14:                                                                                                                                
sycl/plugins/level_zero/../unified_runtime/pi2ur.hpp: In lambda function:                                                                                                                          
sycl/plugins/level_zero/../unified_runtime/pi2ur.hpp:294:7: warning: case value ‘32’ not in enumerated type ‘ur_device_partition_t’ [-Wswitch]                                                     
  294 |       case (ur_device_partition_t)                                                                                                                                                                                      
      |       ^~~~
```
can you advice, how it can be resolved?

Reply:
@smaslov-intel `32` is a value of `UR_DEVICE_AFFINITY_DOMAIN_FLAG_NEXT_PARTITIONABLE`.

> I think the right way to do is to add `default` case that would return an error

There is a `default` case on 298 line which returns an error. How 294-297 lines should be modified? Or maybe removed?

Reply:
ur_device_partition_property_t was before a intptr:

```cpp
///////////////////////////////////////////////////////////////////////////////
/// @brief Device partition property type
typedef intptr_t ur_device_partition_property_t;
```

so the compiler wouldn't complain by casting UR_DEVICE_AFFINITY_DOMAIN_FLAG_NEXT_PARTITIONABLE into ur_device_partition_property_t , but now you are casting it to ur_device_partition_t, which is an enum:

```
typedef enum ur_device_partition_t {
    UR_DEVICE_PARTITION_EQUALLY = 0x1086,            ///< Partition Equally
    UR_DEVICE_PARTITION_BY_COUNTS = 0x1087,          ///< Partition by counts
    UR_DEVICE_PARTITION_BY_AFFINITY_DOMAIN = 0x1088, ///< Partition by affinity domain
    UR_DEVICE_PARTITION_BY_CSLICE = 0x1089,          ///< Partition by c-slice
    /// @cond
    UR_DEVICE_PARTITION_FORCE_UINT32 = 0x7fffffff
    /// @endcond

} ur_device_partition_t;
```

so that's why the compiler complains. so question would be, why we are considering UR_DEVICE_AFFINITY_DOMAIN_FLAG_NEXT_PARTITIONABLE here, if the cases for device partition are the enums above? maybe we shouldn't be even returning it here 

https://github.com/intel/llvm/blob/f4525e901cd6ab4b8cf114350698d93b75ff5f17/sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_device.cpp#L339

what would be the closest to UR_DEVICE_AFFINITY_DOMAIN_FLAG_NEXT_PARTITIONABLE? UR_DEVICE_PARTITION_BY_AFFINITY_DOMAIN ?